pkg/resourcemanager/mysql/mysqluser/mysqluser_reconcile.go

```diff
@@ -147,22 +149,27 @@ func (s *MySqlUserManager) Ensure(ctx context.Context, obj runtime.Object, opts
 		return false, err
 	}
 
-	// apply roles to user
-	if len(instance.Spec.Roles) == 0 {
-		instance.Status.Message = "No roles specified for user"
-		return false, fmt.Errorf("no roles specified for database user")
+	err = mysql.EnsureUserServerRoles(ctx, db, user, instance.Spec.Roles)
+	if err != nil {
+		err = errors.Wrap(err, "ensuring server roles")
+		instance.Status.Message = err.Error()
+		return false, err
 	}
 
-	err = mysql.GrantUserRoles(ctx, user, instance.Spec.DbName, instance.Spec.Roles, db)
+	warnings, err := mysql.EnsureUserDatabaseRoles(ctx, db, user, instance.Spec.DatabaseRoles)
```

Comment:
Same comment here about if we really know that what is ending up in warnings here is something we're ok classifying as a warning.